Cython/Utility/Coroutine.c

```diff
@@ -301,8 +301,11 @@ static void __Pyx_Generator_Replace_StopIteration(int in_async_gen); /*proto*/
 //@requires: Exceptions.c::GetException
 
 static void __Pyx_Generator_Replace_StopIteration(int in_async_gen) {
-    PyObject *exc, *val, *tb, *cur_exc;
+    PyObject *val, *cur_exc, *new_exc;
+    #if __PYX_LIMITED_VERSION_HEX < 0x030C0000
```

Comment:
I think this just needs the 3.12 special-casing removed (because it isn't really the same) and then it should work...